src/components/Privacy/injectSendSetConsentRequest.js

```diff
@@ -10,12 +10,19 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+import ecidNamespace from "../../constants/ecidNamespace";
+
 export default ({
   createConsentRequestPayload,
   sendEdgeNetworkRequest
 }) => consent => {
   const payload = createConsentRequestPayload();
   payload.setConsent(consent);
+  payload.mergeQuery({
```

Comment:
I thought about doing this in the `onBeforeRequest` lifecycle hook instead, but thought we might not need to query for the ECID in ALL requests.. or do we? @jonsnyder @Aaronius @jileeshs

Reply:
I agree that this would be better put within the identity component.  Currently the query is done at the event level inside "onBeforeEvent", but according to this ticket: https://jira.corp.adobe.com/browse/EXEG-1234, the query can be at the payload level.  I propose adding a "mergeQuery" method to "createRequestPayload", and then within the identity component, it can add the ecid query inside the "onBeforeRequest" method.

Reply:
@jfkhoury Sorry for not answering sooner. My understanding is that any request could result in an ECID being established for the user. If we always want the legacy ECID to be created whenever an ECID is established for a user, then I think we _have_ to request the ECID in all requests as long as the legacy ECID hasn't been established previously.

I agree with what I think is being said though, which is that it's weird that Identity isn't the one responsible for requesting the ECID in this PR. It should be Identity's responsibility and not Privacy's.

Reply:
I see you switched it. Nice. 👍